src/eval/merge.rs

```diff
@@ -361,7 +362,9 @@ pub fn merge(
         }
         // Merge put together the fields of records, and recursively merge
         // fields that are present in both terms
-        (Term::Record(mut m1, attrs1), Term::Record(mut m2, attrs2)) => {
+        (Term::Record(r1), Term::Record(r2)) => {
```

Comment:
Yes, we theoretically could, using record syntax, e.g.:

```rust
(Term::Record(RecordBase { fields: mut m1, attrs: attrs1 }, Term::Record(RecordBase { fields: mut m2, attrs: attrs2 }) => ...
```

But as you can see, the result is significantly more verbose because you need to specify which field names map to which new bindings, since record destructuring isn't based on position. 

Ultimately I think binding to the top-level struct & then accessing its fields is probably simpler.

Reply:
> It also avoids the need to change this part of the code when adding a new field to RecordBase

I see that as a benefit rather than a problem because some call sites may need to be updated as well. But in any case I won't block on that

Reply:
Both approaches have arguments, but one appeal of not doing the destructuring is to declare and use fewer top-level identifiers. If you're giving a name to each subfield anyway `m1, m2, attrs1, attrs2`, IMHO you may as well use destructuring, because as Julien brings up, the soundness of the implementation is more resilient to refactoring.

Reply:
I'm not entirely sure what you mean by "soundness" in this context. As far as I can tell, introducing a new field is a purely additive change that can't break existing behaviour. If the new field is _intended_ to change existing behaviour, then I'd expect a change in the associated tests to go alongside it, which would prompt changes here. All destructuring here would do is require you to add either `, new_field: _` or `, ..` to the destructuring, and if you choose the latter then you're in the same state as you are here. It's entirely possible I'm missing something though.

Either way, I'm glad you mentioned the top-level identifiers, as I actually meant to remove those - they were only there to get things compiling faster. :sweat_smile: I'll do that as part of the updates I'm making now.